Source/JavaScriptCore/runtime/TemporalPlainYearMonth.cpp

```diff
@@ -118,6 +118,94 @@ String TemporalPlainYearMonth::toString(JSGlobalObject* globalObject, JSValue op
     return ISO8601::temporalYearMonthToString(m_plainYearMonth, calendarName);
 }
 
+// https://tc39.es/proposal-temporal/#sec-temporal.plainyearmonth.from
+// https://tc39.es/proposal-temporal/#sec-temporal-totemporalyearmonth
+TemporalPlainYearMonth* TemporalPlainYearMonth::from(JSGlobalObject* globalObject, JSValue item, JSValue optionsValue)
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    // Handle string case first so that string parsing errors (RangeError)
+    // can be thrown before options-related errors (TypeError);
+    // see step 4 of ToTemporalYearMonth
+    TemporalPlainYearMonth* result;
+    bool isString = item.isString();
+
+    if (isString) {
+        auto string = item.toWTFString(globalObject);
+        RETURN_IF_EXCEPTION(scope, { });
+        result = TemporalPlainYearMonth::from(globalObject, string);
+        RETURN_IF_EXCEPTION(scope, { });
+        // See step 11 of ToTemporalYearMonth
+        if (optionsValue) {
+            toTemporalOverflow(globalObject, optionsValue);
+            RETURN_IF_EXCEPTION(scope, { });
+        }
+        RELEASE_AND_RETURN(scope, result);
+    }
+
+    JSObject* options = nullptr;
+    if (optionsValue) {
```

Comment:
`optionsValue` is now just `JSValue`, so `if (optionsValue)` means checking if `optionsValue` is not zero value.
But I think what we should do here is checking if `optionsValue` is not `undefined` (or `null`) ?

Reply:
If my comment makes sense, we should add more tests to check this behavior